src/clients/N3Client.ts

```diff
@@ -319,6 +320,15 @@ export class N3Client {
     // return the little endian version of the address
     return u.reverseHex(scriptHash);
   }
+
+  public async retrieveNEP17Info(address: string) {
+    const fn = (q:string): Promise<InvokeResult> => this.rpcClient.invokeFunction(address, q);
+    const key = ['name', 'symbol', 'decimals']
+    const xs: Promise<InvokeResult>[] = key.map(fn);
+    const response = (await Promise.all(xs)).map(({stack:[{type, value}]})=> type === 'Integer' ? value : (typeof value === 'string' && atob(value)))
+    const result = response.reduce((acc, curr, i) => ({ ...acc, [key[i]]: curr }),{ address })
+    return result
+  }
 }
```

Comment:
1. Line 328 is messy and error-prone, so I recommend breaking down and processing the different variables
```
// use optional chaining to avoid errors like cannot read type of undefined
const decimals = typeof response[2]?.stack?.[0]?.value ...
const symbol = ...
```
2. Seems like you can't invoke function for `name`, so you'll probably have to do minimum 2 queries: 1) call `getContractState` to get contract name and 2) batched `invokeScript` to query symbol and decimals (will send img via telegram to batch)

3. Received an error `Function atob is not defined` when I was testing this function, so probably need to replace it. The goal is to convert base64 str => ascii str, so you can use either `Buffer` or the `u.HexString` to do the conversion instead

4. Convert decimals to `number`